webui/react/src/components/ModelVersionDeleteModal.tsx

```diff
@@ -0,0 +1,47 @@
+import { Modal } from 'components/kit/Modal';
+import { paths } from 'routes/utils';
+import { deleteModelVersion } from 'services/api';
+import { ErrorLevel, ErrorType } from 'shared/utils/error';
+import { routeToReactUrl } from 'shared/utils/routes';
+import { ModelVersion } from 'types';
+import handleError from 'utils/error';
+
+interface Props {
+  modelVersion: ModelVersion;
+}
+
+const ModelVersionDeleteModal = ({ modelVersion }: Props): JSX.Element => {
+  const handleOk = async () => {
+    if (!modelVersion) return Promise.reject();
+
+    try {
+      await deleteModelVersion({
+        modelName: modelVersion.model.name ?? '',
+        versionNum: modelVersion.version ?? 0,
+      });
+      routeToReactUrl(paths.modelDetails(String(modelVersion.model.id)));
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Please try again later.',
+        publicSubject: `Unable to delete model version ${modelVersion.version}.`,
+        silent: false,
+        type: ErrorType.Server,
+      });
+    }
+  };
+
+  return (
+    <Modal
+      danger
+      size="small"
+      submit={{ handler: handleOk, text: 'Delete Version' }}
+      title="Confirm Delete Model Version">
+      <div>
+        Are you sure you want to delete &quot; V{modelVersion.version}&quot; from this model?
```

Comment:
For deleting model version, let's write it `"Version ${modelVersion.version}"` for consistency with past modal.  If I give the model version a custom name, then "Version 1" and the name are more prominent than "V1".

<img width="750" alt="Screen Shot 2023-03-30 at 10 24 37 AM" src="https://user-images.githubusercontent.com/643918/228885933-726fbd04-8d84-49a7-95de-62921eba7ef7.png">